### PR TITLE
[SOAR-9830] Check coverage threshold inside a GHA action

### PR DESCRIPTION
**Pull request by jkaluza-rapid7**
### Description

Describe the proposed changes:

  - Add a test coverage threshold check in github workflows

### Testing

- [x] tested in the testbed repository
- [x] tested in the insightconnect-plugins repository

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -36,6 +39,12 @@ jobs:
         else
             for i in "${array[@]}"
             do
-              python -m coverage report --data-file=$i
+                python -m coverage report --data-file=$i
+                percentage=$(python -m coverage report --data-file=$i | grep TOTAL | awk '{print $NF}')
+                number=$(echo $percentage |  sed -r 's/[%]+//g')
```
Comment (igorski-r7) [inline .github/workflows/unit_test.yml:44]:
```suggestion
                number=$(echo $percentage | sed -r 's/[%]+//g')
```

**Review by igorski-r7 — APPROVED** [review]

**Review by llaszuk-r7 — APPROVED** [review]
LGTM 
I tested it on my branch, works very well

File: .github/workflows/unit_test.yml [hunk 2]
```diff
@@ -36,6 +39,12 @@ jobs:
         else
             for i in "${array[@]}"
             do
-              python -m coverage report --data-file=$i
+                python -m coverage report --data-file=$i
+                percentage=$(python -m coverage report --data-file=$i | grep TOTAL | awk '{print $NF}')
+                number=$(echo $percentage | sed -r 's/[%]+//g')
+                if [ $COVERAGE_THRESHOLD -gt "$number" ]; then
+                    echo "Coverage of $percentage is too low for $i"
```
Comment (mberezin-r7) [inline .github/workflows/unit_test.yml:46]:
```suggestion
                    echo "Unit test coverage of $percentage is below the required threshold $COVERAGE_THRESHOLD for $i"
```

**Review by mberezin-r7 — COMMENTED** [review]
Functionally looks good to me. I only suggest to fix the grammar of the fail message. It would also be great if you cold replace the variable `i` with a more descriptive name to tell us what is in the `array` (which could also use a better variable name 🙂).

**Pull request merged**